File: tests/fixtures/edgecase_repo/javascript/src/module2.js
const { Service1 } = require("./module1");

class Base2 {
  constructor(name) {
    this.name = name;
  }
}

class Service2 extends Base2 {
  handle(payload) {
    this.validate(payload);
    return new Service1("x").handle(payload);
  }

  validate(payload) {
    return payload !== null;
  }
}

function helper2(value) {
  return value;
}

module.exports = { Service2 };

File: tests/fixtures/edgecase_repo/rust/src/module2.rs
use crate::module1::Service1;

pub struct Base2 {
    pub name: String,
}

pub struct Service2 {
    pub base: Base2,
}

pub trait Handler2 {
    fn handle(&self, payload: &str) -> bool;
}

impl Handler2 for Service2 {
    fn handle(&self, payload: &str) -> bool {
        self.validate(payload);
        true
    }
}

impl Service2 {
    fn validate(&self, payload: &str) {}
}

pub fn helper_2(value: &str) -> &str {
    value
}
